torch/fx/experimental/GraphManipulation.py

```diff
@@ -61,18 +69,114 @@ def get_size_of_node(fx_module: GraphModule, node: Node) -> size_bytes:
     else:
-        raise RuntimeError('Node has no dtype attr')
+        raise RuntimeError("Node has no dtype attr")
     total_size = size_per_elem_bytes * total_num_of_elems
     output_size = size_per_elem_bytes * output_elem
     return size_bytes(output_size, total_size)
+
+
+def serialize_shape(shape: torch.Size) -> str:
+    shape_str = "["
+    for dim in shape:
+        shape_str += str(dim) + ", "
+    shape_str = shape_str[:-2] + "]"
+    return shape_str
+
+
+def serialize_tensor_quantization(tensor: torch.Tensor) -> Dict:
+    scheme = {}
+    if tensor.is_quantized:
+        scheme["q_scheme"] = torch.qscheme
+        ##TODO finish quantized serialization
+    return scheme
+
+
+def serialize_weight(tensor: torch.Tensor) -> Dict:
+    weight = {}
+    weight["type"] = str(tensor.dtype)
+    weight["is_quantized"] = tensor.is_quantized
+    if tensor.is_quantized:
+        weight["quantized_type"] = serialize_tensor_quantization(tensor)
+    weight["shape"] = serialize_shape(tensor.shape)
+    return weight
+
+
+def serialize_module(fx_module: GraphModule, weights, name_prefix="") -> Dict:
+    serialized_dict = {}
+    serialized_dict["modules"] = {}
+    serialized_dict["weights"] = {}
+    serialized_dict["nodes"] = []
+    parameters = fx_module.named_parameters()
+    for name, p in parameters:
+        if isinstance(p, torch.Tensor):
+            weight = serialize_weight(p)
+            prefix = ""
+            if name_prefix:
+                prefix = name_prefix + "."
+            serialized_dict["weights"][prefix + name] = weight
+            weights[prefix + name] = p
+    for node in fx_module.graph.nodes:
+        node_rep = {}
+        # Get shape/type info, currently not needed for call_module.
+        if node.op != "call_module":
+            shape = getattr(node, "shape", None)
+            if shape:
+                node_rep["shape"] = serialize_shape(shape)
+            else:
+                raise RuntimeError("Node has no shape attr")
+            dtype = getattr(node, "dtype", None)
+            if dtype:
+                node_rep["type"] = str(dtype)
+            else:
+                raise RuntimeError("Node has no dtype attr")
+
+        # Recurse down into any submodules we are calling.
+        if node.op == "call_module":
+            if isinstance(getattr(fx_module, node.target), torch.fx.GraphModule):
+                output = serialize_module(
+                    getattr(fx_module, node.target), weights, node.target
+                )
+                serialized_dict["modules"][node.target] = output
+
+        if node.op == "call_function":
+            node_rep["target"] = str(node.target.__name__)
+        else:
+            node_rep["target"] = str(node.target)
+
+        # Make sure we capture all constants.
+        if node.op == "get_attr":
+            target = getattr(fx_module, node.target)
+            if isinstance(target, torch.Tensor):
```

Comment:
I think this would serialize `Parameters` accessed via `get_attr` redundantly to the code at line `121`. Not necessarily a correctness issue, but it might be prudent to add a check

```
            target = getattr(fx_module, node.target)
            prefix = f'{name_prefix}.' if name_prefix else ''
            qualname = prefix + target
            if isinstance(target, torch.Tensor) and qualname not in weights:
              <...>
```